src/librustc/ty/maps.rs

```diff
@@ -205,7 +206,9 @@ pub struct CycleError<'a, 'tcx: 'a> {
 }
 
 impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
-    pub fn report_cycle(self, CycleError { span, cycle }: CycleError) {
+    pub fn report_cycle(self, CycleError { span, cycle }: CycleError)
```

Comment:
Wait, can this and `CycleError` be private? The only reason they're public is for `try_get`, I think?
Also, `CycleError` can probably be decomposed into its fields now? Not sure though.